tardis/utilities/executors/sshexecutor.py

```diff
@@ -1,17 +1,37 @@
+from typing import Optional
 from ...configuration.utilities import enable_yaml_load
 from ...exceptions.executorexceptions import CommandExecutionFailure
 from ...interfaces.executor import Executor
 from ..attributedict import AttributeDict
 
 import asyncio
 import asyncssh
+import asyncstdlib as a
+
+
+async def probe_max_session(connection: asyncssh.SSHClientConnection):
+    """
+    Probe the sshd `MaxSessions`, i.e. the multiplexing limit per connection
+    """
+    sessions = 0
+    async with a.ExitStack() as aes:
+        try:
+            while True:
+                await aes.enter_context(await connection.create_process())
+                sessions += 1
+        except asyncssh.ChannelOpenError:
+            pass
+    return sessions
```

Comment:
Is there any reason why you are not using `connection.create_session()` instead of using an async context manager and `asyncstdlib`. I am not against it, but would like to understand the reason.

Reply:
Okay, that is good reason. Fine with me to keep this code.